components/search/src/error.rs

```diff
@@ -0,0 +1,31 @@
+/// The error type for all Search component operations. These errors are
+/// exposed to your application, which should handle them as needed.
+use error_support::{ErrorHandling, GetErrorHandling};
+
+/// A list of errors that are internal to the component. This is the error
+/// type for private and crate-internal methods, and is never returned to the
+/// application.
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum Error {
+    #[error("Not Implemented")]
+    NotImplemented(),
+}
+
+// #[non_exhaustive]
```

Comment:
It does look like we can have `[NonExhaustive]` [in UDL](https://github.com/mozilla/uniffi-rs/pull/1924) now!

Reply:
I think that only works with Enums at the moment.

Reply:
Oops, I didn't look carefully enough! 😬